### ISSUE TITLE
Changing autoprefixer options does not invalidate cache

### ISSUE THREAD

**Issue by davej** [gustavnikolaj/express-autoprefixer#15]
So, if I do this:

```js
express.createServer()
    .use(autoprefixer({ browsers: 'last 2 versions', cascade: false }))
    .use(express.static(root))
    .listen(1337);
```

And then load a css file all works well and it autoprefixed correctly. But if I then change the browser query code to this:

```js
express.createServer()
    .use(autoprefixer({ browsers: '> 1%', cascade: false }))
    .use(express.static(root))
    .listen(1337);
```

And reload my browser then the old css code is cached (presumably because the ETAG didn't change). This was confusing and I assumed I was doing something wrong until I dugg into the `express-autoprefixer` code.

**Comment by davej** [comment]
Perhaps express-autoprefixer could add some sort of a serialized version of autoprefixer options to the ETAG?

**Comment by papandreou** [comment]
@davej Nice catch and a good solution suggestion.

**Comment by davej** [comment]
Any movement on this? Or perhaps guidance on implementation if you would prefer a PR.

**Comment by papandreou** [comment]
I guess it's a matter of computing a hash of `args` here: https://github.com/gustavnikolaj/express-autoprefixer/blob/3f8b914ef904265a5ccf390219a606cc7e5dc19f/lib/autoprefixer.js#L11

For example:

```js
var configHash = require('crypto').createHash('sha256').update(args.map(function (arg) {
    return JSON.stringify(arg);
}).join()).digest('hex').slice(0, 10);
```

And then make sure that the hash is checked for here: 
https://github.com/gustavnikolaj/express-autoprefixer/blob/3f8b914ef904265a5ccf390219a606cc7e5dc19f/lib/autoprefixer.js#L34

... and added here where the ETag is issued:
https://github.com/gustavnikolaj/express-autoprefixer/blob/3f8b914ef904265a5ccf390219a606cc7e5dc19f/lib/autoprefixer.js#L65

But then again, how about other sources of autoprefixer configuration that aren't passed in explicitly? Such as usage percentages changing after updating caniuse-db, or the set of browsers to support being defined in package.json or in the `.browserslistrc` file? I guess we'd have to take that into account as well for this mechanism to be bulletproof. Maybe it's possible to instantiate autoprefixer and extract all that config state from it?

**Comment by davej** [comment]
Maybe the simplest interim solution would be an option that allows ETagging to be turned off. My guess is that this module is mostly used in development so the easiest solution might be to just turn it off.

**Comment by papandreou** [comment]
Easiest, yes, but it would be a severe performance regression.

**Comment by davej** [comment]
Yes but users would have to explicitly opt-out of the existing functionality. I guess I'm just not familiar with all the existing inputs to figure out how to serialize them in a deterministic way.

**Comment by papandreou** [comment]
No, me neither. I poked around a bit, but it doesn't seem like autoprefixer exposes the resulting configuration via public properties.

**Comment by davej** [comment]
Would it work if we just generated a new input based on datetime every time the middleware is first initialised? If we mixed this with the existing ETag input then that looks like it might work because it looks like the other inputs can't be updated dynamically without re-initialising the autoprefixer module. 

Of course it would be great if the ETag could persist between new Node instances but this might be a reasonable trade off?
